Bigtable/src/ChunkFormatter.php

```diff
@@ -17,72 +17,119 @@
+ * implements the IteratorAggregate interface, and as such can be iterated over
+ * directly in order to access the formatted rows.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Bigtable\DataClient;
+ *
+ * $dataClient = new DataClient('my-instance', 'my-table');
+ * $formatter = $dataClient->readRows();
+ * ```
  */
-class ChunkFormatter
+class ChunkFormatter implements \IteratorAggregate
 {
-    private $RowStateEnum;
-    private $prevRowKey = null;
+    /**
+     * @var array
+     */
+    private static $rowStateEnum = [
+        'NEW_ROW'          => 1,
+        'ROW_IN_PROGRESS'  => 2,
+        'CELL_IN_PROGRESS' => 3,
+    ];
+
+    /**
+     * @var string
+     */
     private $state;
-    private $serverStream;
-    private $options;
+
+    /**
+     * @var ServerStream
+     */
+    private $stream;
+
+    /**
+     * @var string|null
+     */
+    private $prevRowKey = null;
+
+    /**
+     * @var string|null
+     */
     private $rowKey = null;
+
+    /**
+     * @var array
+     */
     private $row = [];
+
+    /**
+     * @var array|null
+     */
     private $family = null;
+
+    /**
+     * @var array|null
+     */
     private $qualifiers = null;
+
+    /**
+     * @var string|null
+     */
     private $qualifierValue = null;
 
     /**
-     * Constructs ChunkFormatter.
+     * Constructs the ChunkFormatter.
      *
-     * @param \Google\ApiCore\ServerStream $stream input server stream.
-     * @param array $options optional argument.
+     * @param ServerStream $stream A server stream used to read rows.
      */
-    public function __construct($stream, array $options = [])
+    public function __construct(ServerStream $stream)
     {
-        $this->serverStream = $stream;
-        $this->options = $options;
-        $this->RowStateEnum = array(
-            'NEW_ROW' => 1,
-            'ROW_IN_PROGRESS' => 2,
-            'CELL_IN_PROGRESS' =>3,
-        );
+        $this->stream = $stream;
         $this->reset();
     }
 
     /**
-     * Formats the row chunks into friendly format. Chunks contain 3 properties:
-     *
-     * `rowContents` The row contents, this essentially is all data pertaining
-     * to a single family.
-     *
-     * `commitRow` This is a boolean telling us the all previous chunks for this
-     * row are ok to consume.
+     * Formats the row's chunks into a friendly format.
      *
-     * `resetRow` This is a boolean telling us that all the previous chunks are to
-     * be discarded.
+     * Example:
+     * ```
+     * $rows = $formatter->readAll();
+     * ```
      *
-     * @return \Generator|mixed
-     * @throws \Google\Cloud\Bigtable\Exception\BigtableDataOperationException if it detects invalid state
+     * @return \Generator
+     * @throws BigtableDataOperationException If a malformed response is received.
      */
     public function readAll()
     {
-        foreach ($this->serverStream->readAll() as $readRowsResponse) {
+        // Chunks contain 3 properties:
```

Comment:
I moved these notes down into the method body as the docblock gets exposed to users in our doc site, the notes seemed more aimed towards maintainers of the library.